react/src/ReadOnlyMode.js

```diff
@@ -1,17 +1,19 @@
 import Cookie from './utils/Cookie';
 import { defineGlobal, useGlobal } from './Global';
 import Image from './utils/Image';
+import Tooltip from './components/Tooltip';
 
 const _ReadOnlyModeGlobal = defineGlobal(() => Cookie.IsReadOnlyMode());
 const _ReadOnlyModeDisplay = () => {
     const [ readOnlyMode, setReadOnlyMode ] = useGlobal(_ReadOnlyModeGlobal);
-    return <span title={`You are in ${readOnlyMode ? 'readonly' : 'read/write'} mode. Click to enter ${readOnlyMode ? 'read/write' : 'readonly'} mode.`}>
-        <img alt="lock" src={readOnlyMode ? Image.Lock() : Image.Unlock()} style={{height:"35",cursor:"pointer"}}
+    return <>
+        <img id={`tooltip-readonly`} alt="lock" src={readOnlyMode ? Image.Lock() : Image.Unlock()} style={{height:"35",cursor:"pointer"}}
```

Comment:
I'm not sure what's in
```
{`tooltip-readonly`}
```
but is the result of that really suitable to be an id? It feels like it probably wants to be a class. You're using the same quantity as an Id in `Tooltip` farther down. Is that supposed to be the same id? I thought id= specified an element's unique ID so I'm not sure what it means for two elements to have the same id.

Reply:
No the way this works is that the Tooltip element defines an ID which is referenced, exactly, with the same name/ID on the target of the tooltip; so they are paired like this.

Reply:
And both are called `id`? That seems weird, but I guess if that's how it's done.

Reply:
Yeah, that's it ;-/